web/src/main/ui/app/components/flows-new/edit-flow/mapping/ui/mapping-ui.component.ts

```diff
@@ -315,4 +420,31 @@ export class MappingUiComponent implements OnChanges {
     return _.includes(this.targetEntity.definition.primaryKey, name);
   }
 
+  executeFunctions(funcName,propName){
+    this.mapExpresions[propName] = this.mapExpresions[propName] + " " + this.functionsDef(funcName);
+    console.log(funcName,propName,this.mapExpresions[propName])
+  }
+
+  functionsDef(funcName) {
+    return this.mapFunctions[funcName].signature;
+    }
+OpenFullSourceQuery () {
+  let result = this.dialogService.alert(
+    this.step.options.sourceQuery,
+    'OK'
+  );
+  result.subscribe();
+
+}
+
+insertFunction(fname,index) {
+
+    var startPos = this.fieldName.toArray()[index].nativeElement.selectionStart;
```

Comment:
One thing I just noticed - it looks worthwhile to assign "this.fieldName.toArray()[index]" to a variable, as it's used several times below, which makes the code a little harder to read.